projects/python-client/src/datapane/client/api/user.py

```diff
@@ -113,6 +116,76 @@ def _run_script(script: str):
     runpy.run_path(str(script_path), run_name="__datapane__")
 
 
+def _run_template(template_path: str):
+    """Run the template script"""
+    # Store the cwd before we change it
+    old_cwd = os.getcwd()
+    # Change cwd to support relative paths in template script
+    os.chdir(template_path)
+
+    try:
+        runpy.run_path(
+            "template.py",
+            run_name="__datapane__",
+        )
+    # Notify the user of missing packages that are required by the template
+    except ModuleNotFoundError as e:
+        raise DPError(f"Please install the following packages to run this template\n{e.name}") from e
+
+    os.chdir(old_cwd)
+
+
+def _download_template(url: str):
```

Comment:
there is a `URL` type in `datapane.common` afaik

Reply:
Updated to use `URL` from `datapane.common`. It looks like this is something we may need to update as part of a tech-cleanup - many "url" variables are of type `str`. At the moment, `URL` is an alias for `str`.